go-controller/pkg/node/openflow_manager.go

```diff
@@ -245,7 +245,38 @@ func (c *openflowManager) updateBridgeFlowCache(hostIPs []net.IP, hostSubnets []
 	}
 	dftFlows = append(dftFlows, dftCommonFlows...)
 
-	c.updateFlowCacheEntry("NORMAL", []string{fmt.Sprintf("table=0,priority=0,actions=%s\n", util.NormalAction)})
+	if config.OvnKubeNode.Mode == types.NodeModeDPU {
+		// Retrieve the local_ip for the tunnel endpoint which will be used to create OpenFlow actions
+		checkCmd := []string{
+			"get",
+			"Open_vSwitch",
+			".",
+			"external_ids:ovn-encap-ip",
+		}
+		encapIP, _, err := util.RunOVSVsctl(checkCmd...)
+		if err != nil {
+			return err
+		}
+		encapIP = strings.TrimSpace(encapIP)
+
+		// The extra actions, compared to the full mode, are due to a DHCP server on the bridge.
+		// While not impacting the other flows negatively it implies something "custom"
+		// this could be a future feature to allow users to add custom flow actions on the managed ovnkube bridge via Kubernetes API.
+		c.updateFlowCacheEntry("NORMAL", []string{
+			// Add normal action on table 0 (act as a switch)
+			fmt.Sprintf("table=0,priority=0,actions=%s\n", util.NormalAction),
+			// Add OpenFlow action with high priority that allows a normal action for DHCP servers
+			// that are serving on the ovnkube managed bridge.
+			fmt.Sprintf("table=0,priority=999,udp,tp_dst=67,actions=%s\n", util.NormalAction),
+			// Add OpenFlow action with high priority that allows the PF (physical function) to the
+			// local_ip.
+			fmt.Sprintf("table=0,priority=998,ip,nw_dst=%s,actions=%s\n", encapIP, util.NormalAction),
```

Comment:
So with this flow you are trying to ensure that packets from the PF destined to br-ovn IP can be sent. I think you should set action to be LOCAL.

Just to make sure I'm on the same page here. The node IP for the host k8s node is the PF IP right?

Reply:
There is some method behind the NORMAL madness, if enabled, it will help with the MAC learning (FDB entries) on the bridge.

There are good parts about that, it can aid us i.e. for the ARP responder needed for the VTEP intf but also it has its pitfalls i.e.: https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5318

Setting to LOCAL, could work to enable the traffic pattern described above, but I don't know the full impact this will entail because we will use those learned FDB entries.

Just as a reminder in my case I have **two** node IPs, lets name the first one for ease `eth0` and the second the PF `pf0hpf`.

`eth0` - will have the K8s node IP and is our default gw
`pf0hpf` - will be represent our highspeed network